traffic_ops/traffic_ops_golang/ats/atsserver/meta.go

```diff
@@ -85,136 +70,14 @@ func GetConfigMetaData(w http.ResponseWriter, r *http.Request) {
-func getScope(tx *sql.Tx, cfgFile string) (tc.ATSConfigMetaDataConfigFileScope, error) {
-	switch {
-	case cfgFile == "ip_allow.config":
-		fallthrough
-	case cfgFile == "parent.config":
-		fallthrough
-	case cfgFile == "hosting.config":
-		fallthrough
-	case cfgFile == "packages":
-		fallthrough
-	case cfgFile == "chkconfig":
-		fallthrough
-	case cfgFile == "remap.config":
-		fallthrough
-	case strings.HasPrefix(cfgFile, "to_ext_") && strings.HasSuffix(cfgFile, ".config"):
-		return tc.ATSConfigMetaDataConfigFileScopeServers, nil
-	case cfgFile == "12M_facts":
-		fallthrough
-	case cfgFile == "50-ats.rules":
-		fallthrough
-	case cfgFile == "astats.config":
-		fallthrough
-	case cfgFile == "cache.config":
-		fallthrough
-	case cfgFile == "drop_qstring.config":
-		fallthrough
-	case cfgFile == "logs_xml.config":
-		fallthrough
-	case cfgFile == "logging.config":
-		fallthrough
-	case cfgFile == "plugin.config":
-		fallthrough
-	case cfgFile == "records.config":
-		fallthrough
-	case cfgFile == "storage.config":
-		fallthrough
-	case cfgFile == "volume.config":
-		fallthrough
-	case cfgFile == "sysctl.conf":
-		fallthrough
-	case strings.HasPrefix(cfgFile, "url_sig_") && strings.HasSuffix(cfgFile, ".config"):
-		fallthrough
-	case strings.HasPrefix(cfgFile, "uri_signing_") && strings.HasSuffix(cfgFile, ".config"):
-		return tc.ATSConfigMetaDataConfigFileScopeProfiles, nil
-
-	case cfgFile == "bg_fetch.config":
-		fallthrough
-	case cfgFile == "regex_revalidate.config":
-		fallthrough
-	case cfgFile == "ssl_multicert.config":
-		fallthrough
-	case strings.HasPrefix(cfgFile, "cacheurl") && strings.HasSuffix(cfgFile, ".config"):
-		fallthrough
-	case strings.HasPrefix(cfgFile, "hdr_rw_") && strings.HasSuffix(cfgFile, ".config"):
-		fallthrough
-	case strings.HasPrefix(cfgFile, "regex_remap_") && strings.HasSuffix(cfgFile, ".config"):
-		fallthrough
-	case strings.HasPrefix(cfgFile, "set_dscp_") && strings.HasSuffix(cfgFile, ".config"):
-		return tc.ATSConfigMetaDataConfigFileScopeCDNs, nil
-	}
-
-	scope, ok, err := ats.GetFirstScopeParameter(tx, cfgFile)
+	uriSignedDSes, err := ats.GetServerURISignedDSes(inf.Tx.Tx, server.HostName, server.Port)
 	if err != nil {
-		return tc.ATSConfigMetaDataConfigFileScopeInvalid, errors.New("getting scope parameter: " + err.Error())
-	}
-	if !ok {
-		scope = string(tc.ATSConfigMetaDataConfigFileScopeServers)
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("GetConfigMetaData getting server uri-signed dses: "+err.Error()))
+		return
 	}
-	return tc.ATSConfigMetaDataConfigFileScope(scope), nil
+
+	txt := atscfg.MakeMetaConfig(serverName, server, tmParams.URL, tmParams.ReverseProxyURL, locationParams, uriSignedDSes, scopeParams)
+	w.Header().Set(tc.ContentType, tc.ApplicationJson)
+	w.Write([]byte(txt))
+	api.WriteRespRaw(w, r, txt)
```

Comment:
I know `atstccfg` isn't using this API, but I'm pretty sure this line should be removed since we're calling `w.Write` just above it.

Reply:
Yeah, ORT/atstccfg doesn't use this anymore; but still a bug. Fixed.